### PR TITLE
Provide safe wrappers for thread notification API

### PR DESCRIPTION
**Pull request by dcoles**
Furi provides access to [FreeRTOS Task Notification](https://freertos.org/Documentation/02-Kernel/02-Kernel-features/03-Direct-to-task-notifications/01-Task-notifications) feature which can be used for syncronization between threads.

Task notifications act as `u32` bitmask, so can be used to signal multiple potential interesting states on the target thread.

### CODE REVIEW

File: crates/flipperzero/src/furi/thread.rs [hunk 1]
```diff
@@ -233,11 +235,97 @@ pub fn sleep_ticks(duration: Duration) {
...
+impl ThreadId {
+    /// Get the `ThreadId` for the current thread.
+    pub fn current() -> Self {
+        ThreadId(unsafe { sys::furi_thread_get_current_id() })
+    }
+
+    /// Get the `ThreadId` for a specific C `FuriThread`.
+    ///
+    /// # Safety
+    /// The thread pointer must be non-null and point to a valid `FuriThread`.
```
Comment (JarvisCraft) [inline crates/flipperzero/src/furi/thread.rs:250]:
```suggestion
    /// # Safety
    ///
    /// The thread pointer must be non-null and point to a valid `FuriThread`.
```

Minor markdown fix

File: crates/flipperzero/src/furi/thread.rs [hunk 2]
```diff
@@ -233,11 +235,97 @@ pub fn sleep_ticks(duration: Duration) {
 }
 
 /// A unique identifier for a running thread.
-#[cfg(feature = "alloc")]
-#[cfg_attr(docsrs, doc(cfg(feature = "alloc")))]
-#[allow(dead_code)]
+#[derive(Copy, Clone, PartialEq, Eq)]
 pub struct ThreadId(sys::FuriThreadId);
```
Comment (JarvisCraft) [inline crates/flipperzero/src/furi/thread.rs:239]: Let's add `#[repr(transparent)]` to it

File: crates/flipperzero/src/furi/thread.rs [hunk 3]
```diff
@@ -233,11 +235,97 @@ pub fn sleep_ticks(duration: Duration) {
...
+
+/// Wait for up-to `timeout` for a change to any of the specified notification `flags` for the current thread.
+///
+/// If `clear`, then the specified flags will be cleared after a notification is received.
+pub fn wait_any_flags(
+    flags: u32,
+    clear: bool,
+    timeout: Duration,
+) -> Result<u32, sys::furi::Status> {
+    let options = FuriFlag_FuriFlagWaitAny | (if !clear { FuriFlag_FuriFlagNoClear } else { 0 });
```
Comment (JarvisCraft) [inline crates/flipperzero/src/furi/thread.rs:301]:
```suggestion
    let options = FuriFlag_FuriFlagWaitAny | (if clear { 0 } else { FuriFlag_FuriFlagNoClear });
```

Given overall expression complexity, getting rid of inversion seems useful

File: crates/flipperzero/src/furi/thread.rs [hunk 4]
```diff
@@ -233,11 +235,97 @@ pub fn sleep_ticks(duration: Duration) {
...
+
+/// Wait for up-to `timeout` for a change to all of the specified notification `flags` for the current thread.
+///
+/// If `clear`, then the specified flags will be cleared after a notification is received.
+pub fn wait_all_flags(
+    flags: u32,
+    clear: bool,
+    timeout: Duration,
+) -> Result<u32, sys::furi::Status> {
+    let options = FuriFlag_FuriFlagWaitAll | (if !clear { FuriFlag_FuriFlagNoClear } else { 0 });
```
Comment (JarvisCraft) [inline crates/flipperzero/src/furi/thread.rs:319]:
```suggestion
    let options = FuriFlag_FuriFlagWaitAll | (if clear { 0 } else { FuriFlag_FuriFlagNoClear });
```

Same as above

**Review by JarvisCraft — CHANGES_REQUESTED** [review]

**Comment by dcoles** [comment]
@JarvisCraft Thanks for the review. I've accepted your suggestions and added the `repr(transparent)` annotation.

**Review by JarvisCraft — APPROVED** [review]

**Pull request merged**